src/js/components/TextInput/StyledTextInput.js

```diff
@@ -32,8 +32,8 @@ const StyledTextInput = styled.input`
     disabledStyle(
       props.theme.textInput.disabled && props.theme.textInput.disabled.opacity,
     )}
-  ${props => props.theme.textInput && props.theme.textInput.extend};
   ${props => props.textAlign && textAlignStyle}
+  ${props => props.theme.textInput && props.theme.textInput.extend};
```

Comment:
Switching between those lines (35 + 36) might cause a backward compatibility issue.
Regardless, I thought that component props should suppress the theme props, so it would make sense that `props.textAlign` will suppress `props.theme.textInput` IMO.

Reply:
I think `extend` should always come last. If you look at every other usage in Styled*.js in grommet, it comes last. This was the one place it didn't, which didn't look right.